Sources/SpeziLLMOpenAI/LLMOpenAIAuthMiddleware.swift

```diff
@@ -0,0 +1,31 @@
+//
+// This source file is part of the Stanford Spezi open source project
+//
+// SPDX-FileCopyrightText: 2024 Stanford University and the project authors (see CONTRIBUTORS.md)
+//
+// SPDX-License-Identifier: MIT
+//
+
+// Reference: https://github.com/apple/swift-openapi-generator/blob/main/Examples/auth-client-middleware-example/Sources/AuthenticationClientMiddleware/AuthenticationClientMiddleware.swift
+
+import Foundation
+import HTTPTypes
+import OpenAPIRuntime
+
+struct AuthMiddleware: ClientMiddleware {
```

Comment:
Tiny bit of docs would be nice!

Reply:
Addressed in: 55897f80d83ea5f4f883818bafbccc218d563a5a